dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,302 @@
+
+The source code itself is generally optimized to be read and we try to be as clear and self-documenting as possible, erring on the side of verbosity, with numerous comments when needed to explain why code is written as it is. We expect the source to be optimized by tools, so we can keep it understandable by humans.
+
+If anything is unclear or you have questions, reach out on our [Discord channel](https://discord.com/invite/buildWithLit), or ask a question on [GitHub Discussions](https://github.com/lit/lit/discussions). All our communities can be found on the [Lit Community page](https://lit.dev/docs/resources/community/).
+
+# What is lit-html?
+
+lit-html is an HTML templating library. Templates are written in JavaScript by mixing static HTML strings and dynamic JavaScript values using template literals. lit-html enables a functional / UI-as-data programming model, fast initial rendering, and fast updates that minimally update DOM when state changes.
+
+The key feature that enables this is separating static parts of templates from the dynamic parts with template literals, and never traversing or updating the static parts after the initial render. This makes lit-html very performant as can be seen in the [JS Frameworks Benchmark](https://krausest.github.io/js-framework-benchmark/). A great companion to this document is [Justin Fagnani's lit-html talk](https://youtu.be/Io6JjgckHbg).
+
+# Foundational Pieces
+
+lit-html relies on two key platform features for its performance and ergonomics: [JavaScript Tagged Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) and the HTML [`<template>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) element.
+
+## JavaScript Tagged Template Literals
+
+lit-html's rendering process starts with defining a template with a [tagged template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates).
+
+A _tagged_ template literal is essentially a function call that passes the strings literals and the values of expressions to the tag function.
+
+This code:
+
+```js
+let tag = (strings, ...values) => {
+  console.log('strings: ', strings);
+  console.log('values: ', values);
+};
+let text = 'abc';
+tag`<div>${text}</div>`;
+```
+
+prints:
+
+```js
+strings: ["<div>", "</div>", raw: Array(2)]
+values: ["abc"]
+```
+
+A tag function can return any object, it does not have to return a string.
+
+The first argument to the tag is a special array of strings that retains its identity across multiple evaluations of the tagged literal, and is always the exact same array of strings.
+
+We can see that the identity is preserved if we put the tagged literal in a function:
+
+```js
+// Store how many times we've seen each literal
+let counts = new Map();
```

Comment:
This is sort of hard to follow and breaks the flow.  As noted, suggest just removing this section.

Reply:
Done